dataclients/kubernetes/testdata/ingress/ingress-data/ing-with-prule-named-port.yaml

```diff
@@ -41,6 +41,6 @@ subsets:
   - ip: 10.2.9.103
   - ip: 10.2.9.104
   ports:
-  - name: bar
+  - name: baz
```

Comment:
name does not matter as long targetPort is referenced by number to the ep port

Reply:
Isn't this a misconception? I understood it the following way: Kubernetes will reference the service port names, not the targert port names in the endpoint ports. 

So let's say we have the following resources on the cluster

```yaml
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: whoami-deployment
  name: whoami-service
  namespace: default
spec:
  clusterIP: 10.102.22.101
  ports:
  - name: this-is-my-service-port-name
    port: 8080
    protocol: TCP
    targetPort: my-port
  selector:
    app: whoami
  sessionAffinity: None
  type: ClusterIP
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  labels:
    app: whoami
  name: whoami-ingress
  namespace: default
spec:
  rules:
  - host: whoami.test.domain.com
    http:
      paths:
      - backend:
          serviceName: whoami-service
          servicePort: 8080
        pathType: ImplementationSpecific
```

Then the endpoint created by Kubernetes looks like this

```yaml
apiVersion: v1
kind: Endpoints
metadata:
  labels:
    app: whoami-deployment
  name: whoami-service
  namespace: default
subsets:
- addresses:
  - ip: 172.17.0.3
    nodeName: minikube
    targetRef:
      kind: Pod
      name: whoami-deployment-6786bf95fd-fnqnq
      namespace: default
  ports:
  - name: this-is-my-service-port-name
    port: 80
    protocol: TCP
```

Which indeed references the endpoint port by service port name `this-is-my-service-port-name` not the target port name `my-port`.

Unless I am missing something important I think my change is needed to reflect how Kubernetes would create the endpoints in the testcase.

Reply:
thanks reviewing this lead me to write https://github.com/zalando/skipper/issues/1645#issuecomment-742755534

Reply:
Thanks for the writeup!